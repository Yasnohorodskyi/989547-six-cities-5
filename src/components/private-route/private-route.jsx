import React from "react";
import propTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {AuthorizationStatus} from "../../consts";
import {connect} from "react-redux";

const PrivateRoute = (props) => {
  const {render, path, exact, authorizationStatus, type, isUserStatusLoaded} = props;
  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        if (type === `toFavorites`) {
          return (
            authorizationStatus === AuthorizationStatus.AUTH || !(isUserStatusLoaded)
              ? render(routeProps)
              : <Redirect to={`/login`}/>
          );
        } else {
          return (
            authorizationStatus === AuthorizationStatus.AUTH
              ? <Redirect to={`/`}/>
              : render(routeProps)
          );
        }
      }}
    />
  );
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
  isUserStatusLoaded: USER.isUserStatusLoaded
});

PrivateRoute.propTypes = {
  authorizationStatus: propTypes.string.isRequired,
  exact: propTypes.bool.isRequired,
  path: propTypes.string.isRequired,
  render: propTypes.func.isRequired,
  type: propTypes.string.isRequired,
  isUserStatusLoaded: propTypes.bool.isRequired
};

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
