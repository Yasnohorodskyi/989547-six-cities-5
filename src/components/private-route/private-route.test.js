import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import configureStore from "redux-mock-store";

import {PrivateRoute} from "./private-route";

const mockStore = configureStore()({
  USER: {
    authorizationStatus: ``,
    isUserStatusLoaded: true,
  }
});

const noop = () => {};

const authorizationStatus = `NO_AUTH`;
const exact = true;
const path = ``;
const type = ``;
const isUserStatusLoaded = false;

it(`Should private route work`, () => {
  const tree = renderer
  .create(
      <Provider store={mockStore}>
        <MemoryRouter>
          <PrivateRoute
            authorizationStatus={authorizationStatus}
            exact={exact}
            path={path}
            type={type}
            isUserStatusLoaded={isUserStatusLoaded}
            render={noop}
          />
        </MemoryRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
