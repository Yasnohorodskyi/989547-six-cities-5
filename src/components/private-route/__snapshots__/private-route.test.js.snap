// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should private route work 1`] = `null`;
